### ISSUE TITLE
多卡时出错

### ISSUE THREAD

**Issue by liuwanqingqing** [caiyuanhao1998/Retinexformer#168]
你好，我在跑多卡时遇到了这个问题RuntimeError: Expected to have finished reduction in the prior iteration before starting a new one. This error indicates that your module has parameters that were not used in producing loss. You can enable unused parameter detection by passing the keyword argument `find_unused_parameters=True` to `torch.nn.parallel.DistributedDataParallel`, and by  making sure all `forward` function outputs participate in calculating loss. 但是我并没有找到代码中相关位置有这个定义，请问有什么解决方法吗，感谢

**Comment by caiyuanhao1998** [comment]
你好，感谢关注

这个错误通常出现在 torch.nn.parallel.DistributedDataParallel（DDP） 训练时，表示某些参数没有参与计算 loss，导致梯度同步时发生问题。建议你仔细检查一下代码中是否包含了一些 if 语句或者分支没有被计算到。

我试了一下，我这边多卡是可以正常跑的，也可以参考我们提供的 training log 来 debug

如果觉得我们的 repo 有用的话，帮忙点点 star 支持一下吧，感谢 :）
